firmware/ventilator-controller-stm32/Core/Test/Pufferfish/Util/cobs.txt

```diff
@@ -0,0 +1,142 @@
+Scenario: The Util encode_cobs function correctly encodes buffers
+
+  GIVEN("The Util COBS::encode function")
+
+  WHEN("The encoded byte vector is too small to hold the encoded data")
+    THEN("The encode_cobs function reports out_of_bounds status")
+    THEN("The encoded buffer is empty")
+
+  WHEN("The cobs::encode is called on a null byte")
+    THEN("The encode_cobs function reports ok status")
+    THEN("The encoded buffer is as expected '\x01\x01' ")
+
+  WHEN("The cobs::encode is called on a buffer that contains these bytes '0x00 0x00'")
+    THEN("The encode_cobs function reports ok status")
+    THEN("The encoded buffer is as expected '\x01\x01\x01' ")
+
+  WHEN("The cobs::encode function is called on a buffer that contains these bytes '0x11 0x22 0x00 0x33'")
+    THEN("The encode_cobs function reports ok status")
+    THEN("The encoded buffer is as expected '\x03\x11\x22\x02\x33'")
+
+  WHEN("The cobs::encode function is called on a buffer that contains these bytes '0x11 0x22 0x00 0x33' and a non-empty output_buffer is passed")
```

Comment:
I see that you added a test on the independence of cobs::encode vs. the state of output_buffer before the function call, but I don't see a comparable test for cobs::decode yet.

Reply:
completely forgot about decode. will add now